kdTree.js

```diff
@@ -9,52 +9,184 @@
  * @license MIT License <http://www.opensource.org/licenses/mit-license.php>
  */
 
- (function (root, factory) {
-    if (typeof define === 'function' && define.amd) {
-        define(['exports'], factory);
-    } else if (typeof exports === 'object') {
-        factory(exports);
-    } else {
-        factory(root);
-    }
+/*
+function assert(condition, message) {
+	if (condition) {
+		return;
+	}
+	message = message || "Assertion failed";
+	if (typeof Error !== "undefined") {
+		throw new Error(message);
+	}
+	throw message; // Fallback
+}*/
+
+function swap(points, i,j) {
+	const t = points[i];
+	points[i] = points[j];
+	points[j] = t;
+}
+
+// left is always included, right is always excluded,
+// so the range [left,right) contains (right-left) elements.
+function partition(points, left, right, pivIdx, dim) {
```

Comment:
Can you give a detail documentation of this algorithm? Or provide some links about it.

Reply:
This [link](https://www.geeksforgeeks.org/kth-smallestlargest-element-unsorted-array-set-3-worst-case-linear-time/)
provides some information about the algorithm.

Reply:
This is a very standard algorithm whose description you can find in many place. For example wikipedia : https://en.wikipedia.org/wiki/Median_of_medians#Algorithm